satellite/satellitedb/identdb.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package satellitedb
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"strings"
+	"time"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/identity"
+	"storj.io/storj/pkg/storj"
+	dbx "storj.io/storj/satellite/satellitedb/dbx"
+)
+
+type certDB struct {
+	db *dbx.DB
+}
+
+// Set adds a peer identity entry
+func (certs *certDB) Set(ctx context.Context, nodeID storj.NodeID, peerIdent *identity.PeerIdentity) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	tx, err := certs.db.Begin()
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+		} else {
+			err = errs.Combine(err, tx.Rollback())
+		}
+	}()
+
+	if peerIdent == nil {
+		return Error.New("Peer Identity cannot be nil")
+	}
+	chain := identity.EncodePeerIdentity(peerIdent)
+
+	var serialNum []byte
+	query := `SELECT serial_number FROM peer_identities WHERE node_id = ?;`
+	err = tx.QueryRow(certs.db.Rebind(query), nodeID.Bytes()).Scan(&serialNum)
```

Comment:
@zeebo is there a way to use transactions such that the generated methods still work?

Reply:
Yes, the dbx wrapped DB type has a dbx wrapped Tx type. I believe you just call Open.